### ISSUE TITLE
Gear offset state tracked separately for each vehicle

### ISSUE THREAD

**Issue by cgtrout** [BarryCarlyon/FS22_ExtraGears#9]
Hello, thanks for producing this mod.

One thing I'm wondering is whether or not it is possible to have the gear offset saved seperately for each vehicle.  

Example of how this would work: lets say I have vehicles A, B, C and I set A to +6, the gear offset of +6 will only be applied to vehicle A as I flip through the vehicles in the game.  

As it stands the offset is applied globally to each vehicle.

**Comment by BarryCarlyon** [comment]
> As it stands the offset is applied globally to each vehicle.

Well it's more a representation of your current "controller input parameters" rather than modifing the vehicle.

If you are using a shifter and you leave it in "4" and change vehicles then it will be "4" when you switched.
Same would be true if the game allowed you to bind "12" and you leave your input device in "12". (as would be the case if you were using binds rahter than shift binds which this mod also supports, you leave it in 12 and its in 12 when you switch if you had a 12 speed shifter)).

In terms of "realism" when you leave a vehicle you should put the vehicle in neutral before you leave. Then the offset being left as is by your controller is irrelevant.

Same is (likely) true for "vechicle is in reverse of forward when I switch out of it and then my controller inputs are incorrect"

> Example of how this would work: lets say I have vehicles A, B, C and I set A to +6, the gear offset of +6 will only be applied to vehicle A as I flip through the vehicles in the game.

However I think I can easily reset the shift value to 0 on vehicle switch and make it a config option.

However if you switch from A (at +6) to B (now at +0) then back to A it'll still be +0.

Since this is a modification of the "input controller" as apposed to modifiying the vehicles.

I will investigate and see whats best. Right now I'm not "talking" to the vehicles I'm modifying the "game input controls" instead, (at a layer _above_)

Might also be confusing in multiplayer.

As you with a 6 speed shifter will be using shift mode, to get to gear 12 (6 + 6)
Your friend is using "normal" binds as they have a 12 speed, but the warp in and suddenly are in 18 as the vehicle loaded the +6 then read the 12 from the input controller. (12 + 6)
And a third friend is using manual gears, so doesn't need the shift value being read/stored on the vehicle at all. (Well thats a mess, but would probalby wipe the shift modification from the vehicle memory)

So I'll investigate and see what works best if at all.

TLDR: I'll see what I can do. But it might not work as expected since this is a controller modification rather than a vehicle modification.

If we had "force feedback" for gear shifters it would make sense, as you'd warp vehicles and then the controller would move the gear shifter to the gear it was left in by the previour user of the vehicle, which naturally isn't possible.

(And now I'm just rambling)

**Comment by cgtrout** [comment]
Thanks, never considered the multiplayer considerations.  Having the option to reset every time I switch vehicles would be handy.  Also it would be handy to have it reset to 0  for reverse as well as an seperate optional config switch.

**Comment by cgtrout** [comment]
Hi Barry:

I've created a branch on my fork: cgtrout/FS22_ExtraGears that adds the configuration settings "reset on leave" and "reset on direction change".  If you are interested, I can create a pull request.

**Comment by BarryCarlyon** [comment]
I'll take a look.

Currently got a "fix the mod so it works in FS22 v1.2" so when that clears we can look at new featrures
